### PR TITLE
feat(options): adds `changelogTitle` option

### LINKED ISSUE #21: Changelog title
**Issue**:

In my `Changelog.md` I have title but each new version logs are just added on the top of it. Thus title moves at the middle of the file

**Expected**:

I'd like my title to be always on top

**Possible solutions**:

Check if the first line of the file starts from `#` and if it's, add notes under title

### PR DESCRIPTION
**Pull request by stalniy**
Fixes #21

### CODE REVIEW

File: lib/verify.js [hunk 1]
```diff
@@ -4,13 +4,17 @@ const getError = require('./get-error');
...
 module.exports = pluginConfig => {
-  const {changelogFile} = resolveConfig(pluginConfig);
+  const {changelogFile, changelogTitle} = resolveConfig(pluginConfig);
   const errors = [];
 
   if (!isUndefined(changelogFile) && !(isString(changelogFile) && changelogFile.trim())) {
     errors.push(getError('EINVALIDCHANGELOGFILE', {changelogFile}));
   }
 
+  if (changelogTitle && !(isString(changelogTitle) && changelogTitle.trim())) {
```
Comment (pvdlg) [inline lib/verify.js]: Can write this `if` similarly to the one above, using `isUndefined`? I don't remember exactly but I think I encountered a case in which relying on JS falsy conversion created an issue.
```js
!isUndefined(changelogTitle) && !(isString(changelogTitle) && changelogTitle.trim())
```

File: test/verify.test.js [hunk 2]
```diff
@@ -33,3 +33,19 @@ test('Throw SemanticReleaseError if "changelogFile" option is a whitespace Strin
...
+  t.is(error.code, 'EINVALIDCHANGELOGTITLE');
+});
+
+test('Throw SemanticReleaseError if "changelogTitle" option is a whitespace String', t => {
+  const changelogTitle = '  \n \r ';
+  const [error] = t.throws(() => verify({changelogTitle}));
+
+  t.is(error.name, 'SemanticReleaseError');
+  t.is(error.code, 'EINVALIDCHANGELOGTITLE');
+});
```
Comment (pvdlg) [inline test/verify.test.js]: Can you add a test similar to `Throw SemanticReleaseError if "changelogFile" option is a whitespace String` for consistency reasons?

Comment (pvdlg) [inline test/verify.test.js]: Can change the tests:
- `Verify String "changelogFile"` into `Verify String "changelogFile" and "changelogTitle"`
- `Verify undefined "changelogFile"` into `Verify undefined "changelogFile" and "changelogTitle"`

File: lib/definitions/errors.js [hunk 3]
```diff
@@ -8,9 +8,17 @@ module.exports = {
   EINVALIDCHANGELOGFILE: ({changelogFile}) => ({
     message: 'Invalid `changelogFile` option.',
     details: `The [changelogFile option](${linkify(
-      'README.md#changelogfile'
+      'README.md#options'
```
Comment (pvdlg) [inline lib/definitions/errors.js]: This fix and the following one (`assets` => `changelogFile `) are unrelated to this feature. Would you mind moving them in a different commit (that can be on the same PR). Something like ``fix: correct link and description in `EINVALIDCHANGELOGFILE` error``

File: lib/prepare.js [hunk 4]
```diff
@@ -13,6 +13,13 @@ module.exports = async (pluginConfig, notes, logger) => {
     } else {
       logger.log('Create %s', changelogFile);
     }
-    await writeFile(changelogFile, `${notes.trim()}\n${currentFile ? `\n${currentFile}\n` : ''}`);
+
+    const changelog =
+      changelogTitle && currentFile.indexOf(changelogTitle) === 0
```
[the remark this answers is not in the document]
Comment (stalniy) [reply lib/prepare.js]: i will  use `startsWith` if you don’t mind

**Comment by stalniy** [comment]
All makes sense! I will fix soon.

File: test/verify.test.js (hunk 2 above)
Comment (stalniy) [reply test/verify.test.js]: You commented right under that test :)

Comment (stalniy) [reply test/verify.test.js]: done

**Comment by stalniy** [comment]
All fixed, and I created a separate PR #25 for typos

File: test/verify.test.js (hunk 2 above)
Comment (pvdlg) [reply test/verify.test.js]: Oh right sorry. I meant a test similar to `Throw SemanticReleaseError if "changelogFile" option is an empty String`. The reason is that an empty String is falsy, while a white space one is not.

**Comment by pvdlg** [comment]
It's all good except one missing test: https://github.com/semantic-release/changelog/pull/24#discussion_r198172779

File: test/verify.test.js (hunk 2 above)
Comment (stalniy) [reply test/verify.test.js]: Done

**Comment by stalniy** [comment]
Fixed the last one
